### PR TITLE
Added roll batching rule

### PR DESCRIPTION
**Pull request by vfdev-5**
Description:
- Added roll batching rule
- Updated tests

Related to #112

### CODE REVIEW

File: functorch/csrc/BatchRulesViews.cpp [hunk 1]
```diff
@@ -339,6 +350,7 @@ TORCH_LIBRARY_IMPL(aten, FT_BATCHED_KEY, m) {
   VMAP_SUPPORT("squeeze", squeeze_batch_rule);
   VMAP_SUPPORT("squeeze.dim", squeeze_dim_batch_rule);
   VMAP_SUPPORT("_reshape_alias", _reshape_alias_batch_rule);
+  VMAP_SUPPORT("roll", roll_batch_rule);
```
Comment (Chillee) [inline functorch/csrc/BatchRulesViews.cpp:353]: err, is this another one that can be solved by `REDUCTION_BOXED_ARGS`?

Comment (vfdev-5) [reply functorch/csrc/BatchRulesViews.cpp:353]: Naively using `REDUCTION_BOXED_ARGS(roll, 2);` does not work and gives: `RuntimeError: shifts and dimensions must align. shifts: 1, dims:3`

Comment (Chillee) [reply functorch/csrc/BatchRulesViews.cpp:353]: i think the issue comes when we're calling the equivalent of `torch.roll(x, shifts=1, dims=[])`.

The typical behavior for the boxed fallback is to expand `dims` to the entire array.

For example, `torch.sum(x: rank 3 tensor, dim=[])` => `torch.sum(x, dim=[0,1,2])`.

In this case, however, that behavior doesn't work.

So I guess we can't use `REDUCTION_BOXED_ARGS` :)

However, I think there's a case this PR isn't handling correctly either.

```
from functorch import vmap
import torch

def f(x):
    return torch.roll(x, 1)

val = torch.arange(3*4*5).view(3,4,5)
print(f(val[0]))
print(vmap(f)(val)[0])

>>>
tensor([[19,  0,  1,  2,  3],
        [ 4,  5,  6,  7,  8],
        [ 9, 10, 11, 12, 13],
        [14, 15, 16, 17, 18]])
tensor([[59,  0,  1,  2,  3],
        [ 4,  5,  6,  7,  8],
        [ 9, 10, 11, 12, 13],
        [14, 15, 16, 17, 18]])
```

Comment (vfdev-5) [reply functorch/csrc/BatchRulesViews.cpp:353]: I see, thanks a lot for catching that ! I'll figure out how to cover that case (when dims is empty) and update the PR

**Review by Chillee — APPROVED** [review]

**Pull request merged**